### PR TITLE
fix: Assert our overlay sections are current pre-render

### LINKED ISSUE #570: Overlays sometimes render off

### PR DESCRIPTION
**Pull request by kristofbolyai**
fixes https://github.com/Wynntils/Artemis/issues/570

### CODE REVIEW

File: common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java [hunk 1]
```diff
@@ -57,11 +59,40 @@ public static void enableOverlays(List<Overlay> overlays, boolean ignoreState) {
...
+
+        // this assumes sections are equal in size
+        Window window = McUtils.window();
+        SectionCoordinates section = sections.get(0);
+
+        // account for loss of pixels when section does not perfectly divide screen
+        int sqrt = (int) Math.sqrt(OVERLAY_SECTIONS);
+
+        if (Math.abs((section.x2() - section.x1()) * sqrt - window.getGuiScaledWidth()) >= sqrt) {
+            WynntilsMod.error("We sections with wrong width when trying render! Forcing recalculation.");
```
Comment (magicus) [inline common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java]: "This sentence no verb"...?

File: common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java [hunk 2]
```diff
@@ -57,11 +59,40 @@ public static void enableOverlays(List<Overlay> overlays, boolean ignoreState) {
...
+            calculateSections();
+            return;
+        }
+
+        // this assumes sections are equal in size
+        Window window = McUtils.window();
+        SectionCoordinates section = sections.get(0);
+
+        // account for loss of pixels when section does not perfectly divide screen
+        int sqrt = (int) Math.sqrt(OVERLAY_SECTIONS);
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java:81]: Why do we use square roots?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java:81]: 9 sections = 3 sections per row or  per column

Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java:81]: I think that just makes it more complicated and can be simplified.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java:81]: I mean it was like this before

Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java:81]: I'd rather we go
```java
private static final int SECTION_SIDE_NUMBER = 3;

private static final List<SectionCoordinates> sections = new ArrayList<>(SECTION_SIDE_NUMBER * SECTION_SIDE_NUMBER);
```

**Comment by Incompleteusern** [comment]
I think instead of asserting every time prerender we can just hook post `Window#setGuiScale` with DisplayResizeEvent

**Comment by kristofbolyai** [comment]
> I think instead of asserting every time prerender we can just hook post `Window#setGuiScale` with DisplayResizeEvent

How would that fix anything? We already post the event on return

**Comment by Incompleteusern** [comment]
> > I think instead of asserting every time prerender we can just hook post `Window#setGuiScale` with DisplayResizeEvent
> 
> How would that fix anything? We already post the event on return

It detects when the guiScaledWidth/guiScaledHeight are updated, but that already is called by `DisplayResizeEvent` as you mention. I guess it is slightly "closer" to the gui scale modification.

However, this raises the question of how this assertion fixes the problem when the recalculation on resize does not

**Comment by Incompleteusern** [comment]
Furthermore, the titlescreeninit calculation seems unnecessary as `resizeDisplay` is also called during `Minecraft`'s initialization

**Pull request closed without merging**